langchain4j-elasticsearch/src/main/java/dev/langchain4j/store/embedding/elasticsearch/ElasticsearchEmbeddingStore.java

```diff
@@ -280,41 +276,22 @@ public void removeAll(Filter filter) {
         removeByQuery(query);
     }
 
+    /**
+     * The Elasticsearch implementation will simply drop the index instead
+     * of removing all documents one by one.
+     */
     @Override
     public void removeAll() {
```

Comment:
This method should remove all embeddings, but keep the index/collection/storage/etc so that new embeddings can be added in it after removal.
Do I understand correctly that this is not the case for `ElasticsearchKnnEmbeddingStore`?
It would be great if `ElasticsearchEmbeddingStore` would follow the same logic as other stores

Reply:
There's no more `ElasticsearchKnnEmbeddingStore` but a configuration class. So the behavior for `removeAll()` is the same whatever the configuration used (knn or not). 

Side note: if you follow the Elasticsearch best practices, it's always better to have an index template which creates the desired index with the right settings/mapping when the first document is indexed.

That's why "dropping" entirely an index is ok. And when a new document is going to be added to the embeddings store, Elasticsearch will automatically create the index. No needs to keep it around. 

Another solution, if you want to mimic the previous behavior but without sacrificing the performance and the disk usage, would be to:

* drop the index (currently implemented that way)
* create a new empty index, basically it would change the code by:

```java
client.indices().delete(dir -> dir.index(indexName));
client.indices().create(dir -> dir.index(indexName));
```

But that's useless IMHO.

Reply:
Ok, I see. If index is going to be created automatically, then I guess it is fine (you know better and I trust you :))